va/drm/va_drm_utils.c

```diff
@@ -110,6 +112,14 @@ VA_DRM_GetDriverName(VADriverContextP ctx, char **driver_name_ptr, int candidate
         return VA_STATUS_ERROR_INVALID_DISPLAY;
 
     *driver_name_ptr = va_DRM_GetDrmDriverName(drm_state->fd);
+
+    /* Map vgem to WSL2 for Windows subsystem for linux */
+    struct utsname sysinfo = {};
+    if (!strncmp(*driver_name_ptr, "vgem", 4)
+        && !uname(&sysinfo)
+        && strstr(sysinfo.release, "WSL"))
+        *driver_name_ptr = strdup("WSL");
+
     if (!*driver_name_ptr)
```

Comment:
Neither the win32 backend nor the frontend are calling this API. So perhaps we should step back and cover

Why is this grouped with libva-win32 fixes and not a separate PR and _how_ this code gets reached.

Reply:
I'll add more info to the commit message. This gets called when initializing VA with DRM in [WSL](https://learn.microsoft.com/en-us/windows/wsl/about)

You're right, this may be worth a different PR. It's fixing the driver loading for d3d12 but in WSL/DRM, not libva-win32. I'll split out the commits in a second.

Reply:
va_drm_utils: Map d3d12 gallium driver for Windows Subsytem for Linux #688